### PR TITLE
Adapt `high-availability` settings for prow cluster workloads

### PR DESCRIPTION
**Pull request by oliver-goetz**
/kind enhancement

**What this PR does / why we need it**:
Our prow clusters are multi zonal clusters for a while now. 
Thus, this PR adapts high-availability settings of the prow cluster workloads. These are the new settings:
- all `Deployments`/`Statefulsets` with more than one replica are using `topologySpreadConstraints` with `topologyKey=topology.kubernetes.io/zone`now
- `Services` pointing to these apps are annotated with `service.kubernetes.io/topology-aware-hints: "auto"`
- `replicaCount` of `ingress-nginx` is increased to 3. Most HTTP traffic goes to `deck` and `hook` which are using three replicas too. The nodes for system-components have enough memory and CPU, so we could save a bit of cross-zonal traffic.

Additionally, `ingress-nginx` and `oauth2-proxy` are updated to their latest versions and a copy&paste error in `matchExpressions` of `oauth2-proxy` was fixed.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: config/prow/cluster/oauth2-proxy/helm/values.yaml [hunk 1]
```diff
@@ -17,25 +17,29 @@ affinity:
...
+topologySpreadConstraints:
+- maxSkew: 1
+  topologyKey: topology.kubernetes.io/zone
+  whenUnsatisfiable: DoNotSchedule
+  labelSelector:
+    matchExpressions:
+    - key: app.kubernetes.io/name
+      operator: In
+      values:
+      - oauth2-proxy
```
Comment (ScheererJ) [inline config/prow/cluster/oauth2-proxy/helm/values.yaml:30]: Why did you change `ingress-nginx` to `oauth2-proxy`?

File: config/prow/cluster/oauth2-proxy/helm/values.yaml [hunk 2]
```diff
@@ -17,25 +17,29 @@ affinity:
...
+    - key: app.kubernetes.io/instance
+      operator: In
+      values:
+      - oauth2-proxy
```
Comment (ScheererJ) [inline config/prow/cluster/oauth2-proxy/helm/values.yaml:34]: Why did you change `ingress-nginx` to `oauth2-proxy`?

File: config/prow/cluster/oauth2-proxy/helm/values.yaml [hunk 3]
```diff
@@ -17,25 +17,29 @@ affinity:
...
+    - key: app.kubernetes.io/component
+      operator: In
+      values:
+      - authentication-proxy
```
Comment (ScheererJ) [inline config/prow/cluster/oauth2-proxy/helm/values.yaml:38]: Why did you change `controller` to `authentication-proxy`?

File: config/prow/cluster/oauth2-proxy/oauth2-proxy-deployment.yaml [hunk 4]
```diff
@@ -192,35 +205,34 @@ spec:
...
+              values:
+              - oauth2-proxy
+            - key: app.kubernetes.io/instance
+              operator: In
+              values:
+              - oauth2-proxy
+            - key: app.kubernetes.io/component
+              operator: In
+              values:
+              - authentication-proxy
```
Comment (ScheererJ) [inline config/prow/cluster/oauth2-proxy/oauth2-proxy-deployment.yaml:224]: Same as above.

File: config/prow/cluster/oauth2-proxy/helm/values.yaml (hunk 1 above)
Comment (oliver-goetz) [reply config/prow/cluster/oauth2-proxy/helm/values.yaml:30]: It was a copy&paste error, nobody noticed before 😅

File: config/prow/cluster/oauth2-proxy/helm/values.yaml (hunk 2 above)
Comment (oliver-goetz) [reply config/prow/cluster/oauth2-proxy/helm/values.yaml:34]: same here

File: config/prow/cluster/oauth2-proxy/helm/values.yaml (hunk 3 above)
Comment (oliver-goetz) [reply config/prow/cluster/oauth2-proxy/helm/values.yaml:38]: and here

File: config/prow/cluster/oauth2-proxy/oauth2-proxy-deployment.yaml (hunk 4 above)
Comment (oliver-goetz) [reply config/prow/cluster/oauth2-proxy/oauth2-proxy-deployment.yaml:224]: and here

**Review by ScheererJ — APPROVED** [review]
/lgtm
